File: app/src/main/java/com/aayush/filmipedia/util/adapter/BottomNavAdapter.java
package com.aayush.filmipedia.util.adapter;

import java.util.ArrayList;
import java.util.List;

import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;

public class BottomNavAdapter extends SmartFragmentStatePagerAdapter {
    private final List<Fragment> fragments;

    public BottomNavAdapter(FragmentManager fragmentManager) {
        super(fragmentManager);
        fragments = new ArrayList<>();
    }

    public void addFragment(Fragment fragment) {
        fragments.add(fragment);
    }

    @Override
    public Fragment getItem(int position) {
        return fragments.get(position);
    }

    @Override
    public int getCount() {
        return fragments.size();
    }
}

File: app/src/main/java/com/aayush/filmipedia/util/adapter/BottomNavAdapter.kt
package com.aayush.filmipedia.util.adapter

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager

class BottomNavAdapter(fragmentManager: FragmentManager): SmartFragmentStatePagerAdapter(fragmentManager) {
    private val fragments = mutableListOf<Fragment>()

    fun addFragment(fragment: Fragment) = fragments.add(fragment)

    override fun getItem(position: Int) = fragments[position]

    override fun getCount() = fragments.size
}